tests/test_resolve_command.rs

```diff
@@ -414,3 +410,164 @@ fn test_file_vs_dir() {
 
     "###);
 }
+
+#[test]
+fn test_multiple_conflicts() {
+    let mut test_env = TestEnvironment::default();
+    test_env.jj_cmd_success(test_env.env_root(), &["init", "repo", "--git"]);
+    let repo_path = test_env.env_root().join("repo");
+
+    create_commit(
+        &test_env,
+        &repo_path,
+        "base",
+        &[],
+        &[("file1", "first base\n"), ("file2", "second base\n")],
+    );
+    create_commit(
+        &test_env,
+        &repo_path,
+        "a",
+        &["base"],
+        &[("file1", "first a\n"), ("file2", "second a\n")],
+    );
+    create_commit(
+        &test_env,
+        &repo_path,
+        "b",
+        &["base"],
+        &[("file1", "first b\n"), ("file2", "second b\n")],
+    );
+    create_commit(&test_env, &repo_path, "conflict", &["a", "b"], &[]);
+    // Test the setup
+    insta::assert_snapshot!(get_log_output(&test_env, &repo_path), @r###"
+        @   conflict
+        |\  
+        o | b
+        | o a
+        |/  
+        o base
+        o 
+        "###);
+    insta::assert_snapshot!(
+    std::fs::read_to_string(repo_path.join("file1")).unwrap()
+        , @r###"
+    <<<<<<<
+    %%%%%%%
+    -first base
+    +first a
+    +++++++
+    first b
+    >>>>>>>
+    "###);
+    insta::assert_snapshot!(
+    std::fs::read_to_string(repo_path.join("file2")).unwrap()
+        , @r###"
+    <<<<<<<
+    %%%%%%%
+    -second base
+    +second a
+    +++++++
+    second b
+    >>>>>>>
+    "###);
+
+    let editor_script = test_env.set_up_fake_editor();
+
+    // Check that we can manually pick which of the conflicts to resolve first
+    std::fs::write(&editor_script, "expect\n\0write\nresolution file2\n").unwrap();
+    test_env.jj_cmd_success(&repo_path, &["resolve", "file2"]);
+    insta::assert_snapshot!(test_env.jj_cmd_success(&repo_path, &["diff", "-r", "conflict"]), 
+    @r###"
+    Resolved conflict in file2:
+       1     : <<<<<<<
+       2     : %%%%%%%
+       3    1: -secondresolution basefile2
+       4     : +second a
+       5     : +++++++
+       6     : second b
+       7     : >>>>>>>
+    "###);
+    insta::assert_snapshot!(test_env.jj_cmd_success(&repo_path, &["status"]), 
+    @r###"
+    Parent commit: 9a25592b7aee a
+    Working copy : 00dac0bec4b1 conflict
+    Working copy changes:
+    M file1
```

Comment:
@martinvonz I find it surprising that `file1` shows up here. Should this be considered a bug in `jj status`?

I would guess it represents the fact that `jj` put a materialized conflict version of a file there that is different from the Platonic ideal of a conflict that's stored in `jj`'s store. `jj diff` seems to know to ignore these.

Reply:
That confused me quite a lot and I spent quite some time troubleshooting before I figure out what's going it. It turns out that `jj status` diffs compared to the first parent instead of the auto-merged parents like all other commands. I guess I forgot to update it once I decided that comparing to the auto-merged parents was the right thing to do. I'll work on a PR to fix that as soon as I can.

Reply:
Ah, that makes a lot of sense. Thanks for looking into it. I don't know if this issue is urgent to fix, but it's good to know it's there.

Reply:
#846 fixes it. It was such an easy fix that I just wanted it done.